apps/auctions/volume_matching.py

```diff
@@ -0,0 +1,243 @@
+"""
+Original code was written by Harjasleen Malvai
+Volume Matching Auction : buy and sell orders are matched only on volume while price is determined by reference to some external market.
+
+"""
+import asyncio
+import logging
+from honeybadgermpc.preprocessing import (
+    PreProcessedElements as FakePreProcessedElements,
+)
+from honeybadgermpc.progs.mixins.share_arithmetic import (
+    MixinConstants,
+    BeaverMultiply,
+    BeaverMultiplyArrays,
+)
+from honeybadgermpc.progs.fixedpoint import FixedPoint
+from honeybadgermpc.mpc import TaskProgramRunner
+
+config = {
+    MixinConstants.MultiplyShareArray: BeaverMultiplyArrays(),
+    MixinConstants.MultiplyShare: BeaverMultiply(),
+}
+
+
+async def compute_bids(ctx, balances, bids, price):
+    one = ctx.Share(1)
+    zero = ctx.Share(0)
+    fp_zero = FixedPoint(ctx, zero)
+
+    used_balances = {}
+    for key in balances.keys():
+        used_balances[key] = {
+            "eth": create_clear_share(ctx, 0),
+            "erc20": create_clear_share(ctx, 0),
+        }
+
+    buys = []
+    sells = []
+
+    for i, bid in enumerate(bids):
+        addr, vol = bid
+
+        is_sell = await vol.ltz()
+        is_buy = one - is_sell
+
+        sell_vol = fp_zero - FixedPoint(ctx, await (is_sell * vol.share))
+        buy_vol = FixedPoint(ctx, await (is_buy * vol.share))
+
+        is_sell_vol_valid = one - await balances[addr]["erc20"].lt(
+            sell_vol + used_balances[addr]["erc20"]
+        )
+        is_buy_vol_valid = one - await balances[addr]["eth"].lt(
+            (await buy_vol.__mul__(price)) + used_balances[addr]["eth"]
+        )
+
+        fp_is_sell_vol_valid = FixedPoint(ctx, is_sell_vol_valid * 2 ** 32)
+        fp_is_buy_vol_valid = FixedPoint(ctx, is_buy_vol_valid * 2 ** 32)
+
+        sell_vol = await fp_is_sell_vol_valid.__mul__(sell_vol)
+        buy_vol = await fp_is_buy_vol_valid.__mul__(buy_vol)
+
+        sells.append((addr, sell_vol))
+        buys.append((addr, buy_vol))
+
+        used_balances[addr]["erc20"] = used_balances[addr]["erc20"] + sell_vol
+        used_balances[addr]["eth"] = used_balances[addr]["eth"] + (
+            await buy_vol.__mul__(price)
+        )
+
+    return buys, sells
+
+
+async def volume_matching(ctx, bids):
```

Comment:
This is missing comments explaining what this function interface is. What is "bids" supposed to contain? What is the type of it? Are there preconditions? What function does this compute?

Reply:
If possible try to follow the documentation format guidelines in our contributing guidelines: 
https://github.com/initc3/HoneyBadgerMPC/blob/dev/docs/development/contributing.rst#documentation-conventions